### PR TITLE
Add unit tests for Streaming Xpath

### PR DESCRIPTION
**Pull request by himasha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/core/src/test/java/org/apache/synapse/config/xml/StreamXPathTestCase.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * Tests stream xpath related operations
+ */
+public class StreamXPathTestCase extends TestCase {
+
+    private static final String element = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + "\n" + "<bookstore>\n" + "\n"
+            + "<book category=\"cooking\">\n" + "  <title lang=\"en\">Everyday Italian</title>\n"
+            + "  <author>Giada De Laurentiis</author>\n" + "  <year>2005</year>\n" + "  <price>30.00</price>\n"
+            + "</book>\n </bookstore>";
+
+    private static final String xpath1 = "/bookstore/book/title";
```
Comment (nuwanw) [inline modules/core/src/test/java/org/apache/synapse/config/xml/StreamXPathTestCase.java]: name should be in upper case

Comment (himasha) [reply modules/core/src/test/java/org/apache/synapse/config/xml/StreamXPathTestCase.java]: Fixed with 5744e77

File: modules/core/src/test/java/org/apache/synapse/endpoints/dynamic/InboundEndpointTestCase.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            + "    <send receive=\"receiveSeq\">\n" + "<endpoint>\n"
+            + "            <address uri=\"http://localhost:9000/services/SimpleStockQuoteService\"/>\n"
+            + "        </endpoint>\n" + "    </send>\n" + "</sequence>\n"
+            + "<sequence xmlns=\"http://ws.apache.org/ns/synapse\" name=\"receiveSeq\">\n" + "    <send/>\n"
+            + "</sequence>";
+
+    public void testCreateValidInboundEP() throws Exception {
+        ep = factory.createInboundEndpoint(AXIOMUtil.stringToOM(sampleEP), config);
+        Assert.assertNotNull("Inbound Endpoint is null", ep);
+        Assert.assertEquals(ep.getName().equals("HttpListenerEP1"), true);
```
Comment (a5anka) [inline modules/core/src/test/java/org/apache/synapse/endpoints/dynamic/InboundEndpointTestCase.java]: Shall we change these asserts like below? 
```java
        Assert.assertEquals("message",  "HttpListenerEP1", ep.getName());
```
Let's fix in other places as well.

Comment (himasha) [reply modules/core/src/test/java/org/apache/synapse/endpoints/dynamic/InboundEndpointTestCase.java]: Fixed with f5af1ad

File: modules/core/src/test/java/org/apache/synapse/config/xml/StreamXPathTestCase.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * Tests stream xpath related operations
+ */
+public class StreamXPathTestCase extends TestCase {
+
+    private static final String ELEMENT = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + "\n" + "<bookstore>\n" + "\n"
+            + "<book category=\"cooking\">\n" + "  <title lang=\"en\">Everyday Italian</title>\n"
+            + "  <author>Giada De Laurentiis</author>\n" + "  <year>2005</year>\n" + "  <price>30.00</price>\n"
+            + "</book>\n </bookstore>";
+
+    private static final String XPATH1 = "/bookstore/book/title";
```
Comment (ramith) [inline modules/core/src/test/java/org/apache/synapse/config/xml/StreamXPathTestCase.java]: shoudn't the variable name (or comments) be more descriptive ( than xpath1, xpath2)

Comment (himasha) [reply modules/core/src/test/java/org/apache/synapse/config/xml/StreamXPathTestCase.java]: Here xpath1 and xpath2 refer to random xpath expressions,hence the generic names. Any suggestions if they should be more specific?

**Pull request merged**